go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -25,6 +29,34 @@ const (
 	egressFirewallUpdateError      = "EgressFirewall Rules not correctly updated"
 )
 
+// egressfirewallDNSInfo stores information about the dnsNames used by egressfirewalls
+// this information applies to all egressfirewalls in all namespaces
+type egressfirewallDNSInformation struct {
+	sync.Mutex
+	//map of dnsNames to relevant information
+	dnsNameInfo map[string]*dnsInformation
+}
+
+func newEgressFirewallDNSInfo() egressfirewallDNSInformation {
+	return egressfirewallDNSInformation{
+		dnsNameInfo: make(map[string]*dnsInformation),
+	}
+}
+
+//dnsInformation stores information relating to a dns names
+type dnsInformation struct {
+	// the namespaces that have egressfirewall rules relating to this dnsName
+	// using a map of empty structs becuase the typical operation is search
+	Namespaces sets.String
+	// as is an addressSet of all the IPs the dnsName resolves to accross all nodes
+	as addressset.AddressSet
+
+	// ipNodes is a map of maps it is used to relate the tuple ipAddress,nodeName
```

Comment:
It's not a "map of maps", it's a map of sets. And I don't fully follow on the wording "relate the tuple ipAddress,nodeName", it's a node name to a set of IP addresses right?

Reply:
Actually looking at the code it seems it's an IP address to a set of node names. Conclusion: this needs better wording.

Reply:
updated, let me know if it is still unclear